pkg/controller/registry/grpc/source_test.go

```diff
@@ -0,0 +1,152 @@
+//go:generate counterfeiter -o ../../../fakes/fake_registry_store.go ../../../../vendor/github.com/operator-framework/operator-registry/pkg/registry/interface.go Query
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/registry/resolver"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/fakes"
+	"github.com/operator-framework/operator-registry/pkg/api"
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	opserver "github.com/operator-framework/operator-registry/pkg/server"
+)
+
+func server(store registry.Query, port int) (func(), func()) {
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
+	if err != nil {
+		logrus.Fatalf("failed to listen: %v", err)
+	}
+	s := grpc.NewServer()
+
+	api.RegisterRegistryServer(s, opserver.NewRegistryServer(store))
+
+	serve := func() {
+		if err := s.Serve(lis); err != nil {
+			logrus.Fatalf("failed to serve: %v", err)
+		}
+	}
+
+	stop := func() {
+		s.Stop()
+	}
+
+	return serve, stop
+}
+
+type FakeSourceSyncer struct {
+	// using a map[int] to preserve order
+	History map[resolver.CatalogKey]map[int]connectivity.State
```

Comment:
Question: This might be a dumb question, but for your use-case, what does a `map[int]<t>` give you that a `[]<t>` doesn't?

Reply:
Nothing - no idea why I did this 😄